src/lib/libaddfunction.js

```diff
@@ -149,17 +149,21 @@ addToLibrary({
     if (freeTableIndexes.length) {
       return freeTableIndexes.pop();
     }
+    // record last index
+    var lastUnused =  {{{ from64Expr('wasmTable.length') }}};
     // Grow the table
     try {
+      var growBy = lastUnused;
+      freeTableIndexes.push(...Array.from({ length: growBy - 1 }, (_, i) => lastUnused + i + 1));
```

Comment:
This is probably going to be really expensive for large tables. You might want to track free ranges instead of free indices.

Reply:
This already gone, I have changed it to the suggestions by hoodmane. I will push soon, but I have to fix a failing test.